sequence_processing_pipeline/Pipeline.py

```diff
@@ -703,80 +750,72 @@ def _get_sample_names_from_mapping_file(self, project_name):
-
-        matches = search(r'^(.+)_(\d+)$', str(project_name))
-
-        if matches is None:
-            raise ValueError(f"'{project_name}' does not contain a Qiita-ID.")
+        """
+        # The main functionality of this method has been replaced by this call
+        # to metapool's parse_project_name, but I can't guarantee this function
+        # isn't used somewhere else in the codebase, so I'm not deleting it.
+        proj_info = parse_project_name(project_name)
 
         if short_names is False:
             # return the fully-qualified project name w/Qiita ID.
-            return project_name, matches[2]
+            return project_name, proj_info[QIITA_ID_KEY]
         else:
             # return the project's name and qiita_id
-            return matches[1], matches[2]
+            return proj_info[PROJECT_SHORT_NAME_KEY], proj_info[QIITA_ID_KEY]
 
     def get_project_info(self, short_names=False):
         # test for self.mapping_file, since self.sample_sheet will be
         # defined in both cases.
         results = []
 
+        contains_replicates = None
         if self.mapping_file is not None:
-            if 'contains_replicates' in self.mapping_file:
+            if CONTAINS_REPLICATES_KEY in self.mapping_file:
                 contains_replicates = True
             else:
                 contains_replicates = False
 
             sample_project_map = {pn: _df.sample_name.values for pn, _df in
-                                  self.mapping_file.groupby('project_name')}
-
-            for project in sample_project_map:
-                p_name, q_id = self._parse_project_name(project, short_names)
-                results.append(
-                    {'project_name': p_name, 'qiita_id': q_id,
-                     'contains_replicates': contains_replicates})
+                                  self.mapping_file.groupby(_PROJECT_NAME_KEY)}
+            projects_info = \
+                {p: parse_project_name(p) for p in sample_project_map}
         else:
-            bioinformatics = self.sample_sheet.Bioinformatics
-            for res in bioinformatics.to_dict('records'):
-                p_name, q_id = self._parse_project_name(res['Sample_Project'],
-                                                        short_names)
-
-                # parsed SampleSheet() objects should now contain only
-                # boolean values in contains_replicates column.
-
-                contains_replicates = False
-
-                if 'contains_replicates' in res:
-                    if res['contains_replicates'] is True:
-                        contains_replicates = True
-
-                results.append({'project_name': p_name,
-                                'qiita_id': q_id,
-                                'contains_replicates': contains_replicates})
+            projects_info = self.sample_sheet.get_projects_details()
+        # if this is/isn't a mapping file
```

Comment:
?

Reply:
Sorry, I guess that's my own personal shorthand for "ending the if/else statement checking if this is /else this isn't a mapping file.  I have replaced it with something less cryptic :)